cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/PreventBrokerScaleDownCheck.java

```diff
@@ -0,0 +1,149 @@
+                        LOGGER.debugCr(reconciliation, "Creating AdminClient for cluster {}/{}", reconciliation.namespace());
+                        Admin kafkaAdmin = adminClientProvider.createAdminClient(bootstrapHostname, compositeFuture.resultAt(0), compositeFuture.resultAt(1), "cluster-operator");
+
+                        Future<Set<String>> topicNames = topicNames(kafkaAdmin);
+                        descriptions = topicNames.compose(names -> describeTopics(kafkaAdmin, names));
+                        descriptions
+                                .compose(topicDescriptions -> {
+                                    Set<Integer> idsContainingPartitionReplicas = new HashSet<>();
+                                    // Provides the node IDs which the user would like to remove first
+                                    for (Integer id: idsToBeRemoved)   {
+                                        boolean result =  brokerHasAnyReplicas(reconciliation, topicDescriptions, id);
+                                        if (result) {
+                                            idsContainingPartitionReplicas.add(id);
+                                        }
+                                    }
+                                    cannotScaleDown.complete(idsContainingPartitionReplicas);
+                                    kafkaAdmin.close();
+                                    return cannotScaleDown.future();
+                                }).recover(error -> {
+                                    LOGGER.warnCr(reconciliation, "Failed to get topic descriptions", error);
+                                    cannotScaleDown.fail(error);
+                                    kafkaAdmin.close();
+                                    return Future.failedFuture(error);
+                                });
+
+                    } catch (KafkaException e) {
+                        LOGGER.warnCr(reconciliation, "Failed to check if broker contains any partition replicas", e.getMessage());
+                    }
+
+                    return resultPromise.future();
+                });
+
+        return cannotScaleDown.future();
+    }
+
+    /**
+     * This method gets the topic names after interacting with the Admin client
+     *
+     * @param kafkaAdmin                    Instance of Kafka Admin
+     * @return  return the set of topic names
+     */
+    protected Future<Set<String>> topicNames(Admin kafkaAdmin) {
+        Promise<Set<String>> namesPromise = Promise.promise();
+        kafkaAdmin.listTopics(new ListTopicsOptions().listInternal(true)).names()
+                .whenComplete((names, error) -> {
+                    if (error != null) {
+                        namesPromise.fail(error);
+                    } else {
+                        namesPromise.complete(names);
+                    }
+                });
```

Comment:
Should we have some VertxUtils class method to go from a Kafka future to a Vertx one?